gravitee-am-service/src/main/java/io/gravitee/am/service/impl/PasswordPolicyServiceImpl.java

```diff
@@ -87,20 +87,28 @@ public Single<PasswordPolicy> create(ReferenceType referenceType, String referen
         log.debug("Create a new password policy named '{}' for {} {}", policy.getName(), referenceType, referenceId);
 
         final var entity = policy.toPasswordPolicy(referenceType, referenceId);
-        // TODO during AM-2893, check if there is existing policies, if not set this one as default
 
-        return passwordPolicyRepository.create(entity)
-                .flatMap(createdPolicy -> {
-                    Event event = new Event(Type.PASSWORD_POLICY, new Payload(createdPolicy.getId(), referenceType, referenceId, Action.CREATE));
-                    return eventService.create(event).flatMap(__ -> Single.just(createdPolicy));
+        return passwordPolicyRepository.findByDefaultPolicy(referenceType, referenceId)
+                .flatMap(__ -> {
+                    entity.setDefaultPolicy(Boolean.FALSE);
+                    return Maybe.just(entity);
                 })
-                .doOnSuccess(createdPolicy -> auditService.report(AuditBuilder.builder(PasswordPolicyAuditBuilder.class)
-                        .principal(principal)
-                        .type(EventType.PASSWORD_POLICY_CREATED)
-                        .policy(createdPolicy)))
-                .doOnError(error -> auditService.report(AuditBuilder.builder(PasswordPolicyAuditBuilder.class)
-                        .principal(principal)
-                        .type(EventType.PASSWORD_POLICY_CREATED).throwable(error)));
+                .switchIfEmpty(Single.fromCallable(() -> {
+                    entity.setDefaultPolicy(Boolean.TRUE);
+                    return entity;
+                }))
+                .flatMap(e -> passwordPolicyRepository.create(e)
+                        .flatMap(createdPolicy -> {
+                            Event event = new Event(Type.PASSWORD_POLICY, new Payload(createdPolicy.getId(), referenceType, referenceId, Action.CREATE));
+                            return eventService.create(event).flatMap(___ -> Single.just(createdPolicy));
```

Comment:
___ 

aren't there too many floors?

Reply:
I also think that this is to many floors but I need to first check if in DB is default policy present. I added just one stream above existing one.

Reply:
I mean this "flatMap(___ -> Single.just(createdPolicy));"
"_ _ _" 


why not _________________? :dancers:

Reply:
about flatMap levels:
you don't need "e" object and can do the flat map on "main stream" instead of creating a next level under passwordPolicyRepository.create(e)
 .flatMap(e -> passwordPolicyRepository.create(e)